Sources/BraintreeCore/BTGraphQLHTTP.m

```diff
@@ -63,12 +63,22 @@ - (void)handleRequestCompletion:(NSData *)data
 {
     // Network error
     if (error) {
-        [self callCompletionBlock:completionBlock body:nil response:nil error:error];
+        [self callCompletionBlock:completionBlock body:nil response:(NSHTTPURLResponse *)response error:error];
         return;
     }
 
-    NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
-    BTJSON *body = [[BTJSON alloc] initWithValue:json];
+    BTJSON *body;
+    
+    if (data == nil) {
+        NSError *error = [[NSError alloc] initWithDomain:BTHTTPErrorDomain
+                                                            code:BTHTTPErrorCodeUnknown
+                            userInfo:@{NSLocalizedDescriptionKey: @"An unexpected error occurred with the HTTP request."}];
+        [self callCompletionBlock:completionBlock body:nil response:(NSHTTPURLResponse *)response error:error];
+        return;
+    } else {
+        NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
+        body = [[BTJSON alloc] initWithValue:json];
+    }
```

Comment:
Since the if returns, you can flatten this and treat the `if (data == nil)` line a swift guard
```swift
guard let data = data else {
    // handle error
    return
}
```
```suggestion
    if (data == nil) {
        NSError *error = [[NSError alloc] initWithDomain:BTHTTPErrorDomain
                                                            code:BTHTTPErrorCodeUnknown
                            userInfo:@{NSLocalizedDescriptionKey: @"An unexpected error occurred with the HTTP request."}];
        [self callCompletionBlock:completionBlock body:nil response:(NSHTTPURLResponse *)response error:error];
        return;
    }
    NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
    BTJSON *body = [[BTJSON alloc] initWithValue:json];
```

Purely stylistic, but in my opinion it's a nice benefit to not have to predefined `BTJSON *body;`